### PR TITLE
CB-18485 Modify proxy config of DataHubs

### PR DESCRIPTION
**Pull request by Bajzathd**
See detailed description in the commit message.

### CODE REVIEW

File: environment/src/main/java/com/sequenceiq/environment/environment/service/datahub/DatahubModifyProxyConfigPollerService.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        }
+        if (multipleFlowsResultEvaluator.anyFailed(flowIdentifiers)) {
+            throw new DatahubOperationFailedException("Data Hub modify proxy config error happened. One or more Data Hubs are not modified.");
+        }
+    }
+
+    private FlowIdentifier startModifyProxy(String datahubCrn, String previousProxyConfigCrn) {
+        try {
+            return datahubService.modifyProxy(datahubCrn, previousProxyConfigCrn);
+        } catch (WebApplicationException e) {
```
Comment (bergerdenes) [inline environment/src/main/java/com/sequenceiq/environment/environment/service/datahub/DatahubModifyProxyConfigPollerService.java:76]: It is caught in the DataHub service

File: environment/src/test/java/com/sequenceiq/environment/environment/flow/modify/proxy/handler/ProxyModificationOnDatahubsHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    private static final String ENV_CRN = "env-crn";
+
+    private static final String NAME = "name";
+
+    private static final String PROXY_CRN = "proxy-crn";
+
+    private static final String PREV_PROXY_CRN = "prev-proxy-crn";
+
+    @Mock
+    private DatahubModifyProxyConfigPollerService datahubModifyProxyConfigPollerService;
```
Comment (bergerdenes) [inline environment/src/test/java/com/sequenceiq/environment/environment/flow/modify/proxy/handler/ProxyModificationOnDatahubsHandlerTest.java:41]: Call  to this service mock is not `verify`-ed

**Review by bergerdenes — CHANGES_REQUESTED** [review]
small items

File: core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/endpoint/DistroXV1Endpoint.java [hunk 3]
```diff
@@ -577,4 +578,12 @@ DetachRecipeV4Response detachRecipeByCrn(@ValidCrn(resource = CrnResourceDescrip
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = "validates if the distrox cluster is recoverable or not", nickname = "getClusterRecoverableByCrn")
     RecoveryValidationV4Response getClusterRecoverableByCrn(@ValidCrn(resource = CrnResourceDescriptor.DATAHUB) @PathParam("crn") String crn);
+
+    @POST
```
[the remark this answers is not in the document]
Comment (Bajzathd) [reply core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/endpoint/DistroXV1Endpoint.java:582]: It is PUT in sdx, so I will modify it

**Review by bergerdenes — APPROVED** [review]

**Pull request merged**